src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessWaitState.Unix.cs

```diff
@@ -582,6 +584,21 @@ private bool TryReapChild(bool configureConsole)
                 {
                     // Process is still running
                 }
+                else if (s_allowReapingByOthers && Interop.Sys.GetLastError() == Interop.Error.ECHILD)
+                {
+                    // Process is reapped by other thread(library). Continue execution.
+                    _exitCode = null;
```

Comment:
We can't determine the exit code.
At this point, you should  assign a non-null integer, which is the value that `Process.ExitCode` will return for this process.